src/nvim/os/stdpaths.c

```diff
@@ -0,0 +1,88 @@
+#include "nvim/os/os.h"
```

Comment:
Perhaps a comment up here mentioning `this implements the XDG specification`, with a link to the spec.